### PR TITLE
[Unity][DistIR] LowerDistIR

### PR DESCRIPTION
**Pull request by jinhongyii**
A pass that lowers DistIR to Relax function that runs on each worker.

### CODE REVIEW

**Comment by jinhongyii** [comment]
cc: @yongwww @tqchen

File: src/relax/distributed/transform/lower_distir.cc [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+  }
+
+  void InputPreprocessing() {
+    for (int i = 0; i < static_cast<int>(func_->params.size()); i++) {
+      Var param = func_->params[i];
+      if (const auto* dtensor_sinfo = GetStructInfoAs<DTensorStructInfoNode>(param)) {
+        EmitBroadcastOrScatter(param, new_params_[i], GetRef<DTensorStructInfo>(dtensor_sinfo));
+      } else if (const auto* tuple_sinfo = GetStructInfoAs<TupleStructInfoNode>(param)) {
+        for (int j = 0; j < static_cast<int>(tuple_sinfo->fields.size()); j++) {
+          if (const auto* dtensor_sinfo = tuple_sinfo->fields[j].as<DTensorStructInfoNode>()) {
```
Comment (yongwww) [inline src/relax/distributed/transform/lower_distir.cc:162]: perhaps we can add a Warning or throw an error for non-DTensorStructInfoNode elements in tuple

**Review by yongwww — APPROVED** [review]
LGTM, left one minor comment

File: src/relax/distributed/transform/lower_distir.cc (hunk 1 above)
Comment (jinhongyii) [reply src/relax/distributed/transform/lower_distir.cc:162]: It's possible that the input contains a non-DTensorStructInfoNode element. For example, kv cache is object.

**Pull request merged**